common/scala/src/main/scala/whisk/core/connector/Message.scala

```diff
@@ -147,7 +147,7 @@ case class Activation(name: String,
                       kind: String,
                       conductor: Boolean,
                       memory: Int,
-                      causedBy: Boolean)
+                      causedBy: Option[String])
```

Comment:
this would mean if we serialise the metric for a non-sequence, we won't have the variable in json, right?
any benefits apart from reducing the payload size?

Reply:
may not be relevant: causedBy was used to indicate an activation caused by a trigger. Then became caused by a sequence (so its absence means this is a standalone singleton invoke). Now it also includes conductor actions.

Reply:
I don't understand this change.

Reply:
Which API? It slightly alters the schema of user-events.

Reply:
I think now it is always defined. I personally would prefer the same approach, as it requires less changes for the metric service.